app/V1Module/presenters/LoginPresenter.php

```diff
@@ -161,4 +161,36 @@ public function actionRefresh() {
     ]);
   }
 
+  public function checkIssueToken() {
+    if (!$this->getAccessToken()->isInScope(AccessToken::SCOPE_MASTER)) {
+      throw new ForbiddenRequestException("Restricted tokens cannot be used to issue new tokens");
+    }
+  }
+
+  /**
+   * Issue a new access token with a restricted set of scopes
+   * @POST
+   * @LoggedIn
+   * @Param(type="post", name="scopes", validation="list", description="A list of requested scopes")
+   * @Param(type="post", required=false, name="expiration", validation="integer", description="How long the token should be valid (in seconds)")
+   * @throws ForbiddenRequestException
+   */
+  public function actionIssueToken() {
```

Comment:
Wouldn't it be better to name this method `issueRestrictedToken`? `issueToken` itself can be maybe mistaken for basic login... ?